packages/editor/src/core/components/hooks/options.ts

```diff
@@ -26,22 +26,17 @@ export const OptionsContext = createContext<Options>({
  */
 export const useOptions = () => useContext(OptionsContext);
 
-export const UiTranslatorContext = createContext<UiTranslator>({
-  t: (label) => label,
-});
-
+export type TranslatorFunction = (key: string) => string;
 /**
  * @returns the an object with a single `t` function for ui translations
  */
-export const useUiTranslator = () => {
-  const userTranslatorFunction = useContext(UiTranslatorContext);
+export const useUiTranslator = (): {
+  t: TranslatorFunction;
+} => {
+  const { uiTranslator } = useOptions();
   return {
     t: (key: string) => {
-      const translation = userTranslatorFunction.t(key);
-      if (translation === undefined || translation === null) {
-        return key;
-      }
-      return translation;
+      return uiTranslator?.(key) ?? key;
```

Comment:
?? does fallback to the right operand if left ist null or undefined. So the result is the same

Reply:
if the left is an empty string, it will go for the key as well?

Reply:
the `??` operator only check for null or undefined I think. So I think it is the same.

Reply:
yes, its the same. As you also correctly noticed that devs might indeed want to use empty strings in certain cases.